src/Libraries/Microsoft.Extensions.AI.AzureAIInference/AzureAIInferenceChatClient.cs

```diff
@@ -317,7 +317,7 @@ private ChatCompletionsOptions ToAzureAIOptions(IList<ChatMessage> chatContents,
                         default:
                             if (prop.Value is not null)
                             {
-                                byte[] data = JsonSerializer.SerializeToUtf8Bytes(prop.Value, JsonContext.GetTypeInfo(prop.Value.GetType(), ToolCallJsonSerializerOptions));
+                                byte[] data = JsonSerializer.SerializeToUtf8Bytes(prop.Value, ToolCallJsonSerializerOptions.GetTypeInfo(typeof(object)));
```

Comment:
Why is this using typeof(object) rather than the actual type of the value?

Reply:
Also, the JsonContext.GetTypeInfo method was first doing the lookup on the supplied JSO and then falling back to doing it on the default JSO. This is only doing it on the supplied one.

Reply:
> Why is this using typeof(object) rather than the actual type of the value?

It's equivalent. Serializing something as object has the serializer use the contract for the runtime type.

> Also, the JsonContext.GetTypeInfo method was first doing the lookup on the supplied JSO and then falling back to doing it on the default JSO. This is only doing it on the supplied one.

I removed that intentionally. Even if `JsonContext` does provide a `JsonTypeInfo<object>`, because of how `object` serialization works it will then immediately try to look up the runtime type from `JsonContext` only. For any type graph that isn't closed (i.e. contains nodes with `object`) we should default to consulting `ToolCallJsonSerializerOptions` so that users can specify serializers for any runtime types that may appear in the objects.